pkg/storage/store.go

```diff
@@ -30,4 +30,42 @@ type Store interface {
 
 	// Get fetches the record based on key
 	Get(k string) ([]byte, error)
+
+	// Iterator returns an iterator for the latest snapshot of the
+	// underlying store
+	//
+	// Args:
+	//
+	// start: Start of the key range, include in the range.
+	// limit: Limit of the key range, not include in the range.
+	//
+	// Returns:
+	//
+	// StoreIterator: iterator for result range
+	Iterator(start, limit string) StoreIterator
+}
+
+// StoreIterator is the iterator for the latest snapshot of the underlying store.
+type StoreIterator interface {
+	// Next moves the iterator to the next key/value pair.
+	// It returns false if the iterator is exhausted.
+	Next() bool
+
+	// Release releases associated resources. Release should always success
+	// and can be called multiple times without causing error.
+	Release()
+
+	// Error returns any accumulated error. Exhausting all the key/value pairs
+	// is not considered to be an error.
+	Error() error
+
+	// Key returns the key of the current key/value pair, or nil if done.
+	// The caller should not modify the contents of the returned slice, and
+	// its contents may change on the next call to any 'seeks method'.
+	Key() []byte
+
+	// Value returns the value of the current key/value pair, or nil if done.
+	// The caller should not modify the contents of the returned slice, and
+	// its contents may change on the next call to any 'seeks method'.
+	Value() []byte
```

Comment:
I would rather create one `Result` function with a signature like this `Result() key,val,err` . In my opinion, this is more clear.

Reply:
oh, I see it is leveldb iterator